components/core/overlay/overlay-position.ts

```diff
@@ -5,7 +5,22 @@
 
 import { ConnectedOverlayPositionChange, ConnectionPositionPair } from '@angular/cdk/overlay';
 
-export const POSITION_MAP: { [key: string]: ConnectionPositionPair } = {
+export type POSITION_TYPE =
+  | 'top'
+  | 'topCenter'
+  | 'topLeft'
+  | 'topRight'
+  | 'right'
+  | 'rightTop'
+  | 'rightBottom'
+  | 'bottom'
+  | 'bottomCenter'
+  | 'bottomLeft'
+  | 'bottomRight'
+  | 'left'
+  | 'leftTop'
+  | 'leftBottom';
+export const POSITION_MAP: { [key in POSITION_TYPE]: ConnectionPositionPair } = {
```

Comment:
You could declare `POSITION_MAP` as const and drop the type annotation here. And `POSITION_TYPE` could be `keyof typeof POSITION_MAP`. It would help you get rid of the trouble of writing literal value of `POSITION_TYPE`.